### DISCUSSION TITLE
YouTube J2ME

### DISCUSSION THREAD

**Discussion by unhealthygrapejuice** [ftde0/yt2009#192] [General]
I've attempted a similar method of trying to decompile or edit the jar file of the minitube.jar apps that YouTube provided, and every time I can't seem to find where it's trying to get the terms of use from? It does have the gdata URLs which I replaced, but I can't fix the terms of use, which prevents you from installing it until you accept. Has anyone actually got further than that?

**Comment by beitanam** [comment]
I was looking into them for quite a while before, and using fiddler i figured they try to go to mobilevideo... which is protobuf.
They dont seem to use gdata api sadly.

**Reply by ljcool2006** [reply]
it seems to use something called MASF

**Comment by gitmanbloggus** [comment]
> I've attempted a similar method of trying to decompile or edit the jar file of the minitube.jar apps that YouTube provided, and every time I can't seem to find where it's trying to get the terms of use from? It does have the gdata URLs which I replaced, but I can't fix the terms of use, which prevents you from installing it until you accept. Has anyone actually got further than that?

ill get looking... howd you decompile it?

**Comment by gitmanbloggus** [comment]
> I was looking into them for quite a while before, and using fiddler i figured they try to go to mobilevideo... which is protobuf.
> They dont seem to use gdata api sadly.

then lets convert jtube to gdata

**Comment by ljcool2006** [comment]
copying the files inside this zip file to the `kemnnmod\rms\WW91VHViZV9Zb3VUdWJl\UHJlZmVyZW5jZXM=` directory will allow you to get through the terms of service:
[bypasstos.zip](https://github.com/user-attachments/files/19933144/bypasstos.zip)

**Reply by gitmanbloggus** [reply]
nice, will try and patch the .jar now

**Comment by R0ci0-V3rd3al** [comment]
wow

**Comment by gitmanbloggus** [comment]
> I've attempted a similar method of trying to decompile or edit the jar file of the minitube.jar apps that YouTube provided, and every time I can't seem to find where it's trying to get the terms of use from? It does have the gdata URLs which I replaced, but I can't fix the terms of use, which prevents you from installing it until you accept. Has anyone actually got further than that?

what jar?

**Reply by ljcool2006** [reply]
https://web.archive.org/web/20120130061158/http://www.google.com:80/m/download/youtube/apps/v1.4.7/L1/minitube-147-m2_L1.jar

**Reply by gitmanbloggus** [reply]
> https://web.archive.org/web/20120130061158/http://www.google.com:80/m/download/youtube/apps/v1.4.7/L1/minitube-147-m2_L1.jar

thank you, will try patching soon
